File: DataBinding/Section/IListBinding/C#/IListBinding.DataLayer/Product.cs

namespace GrapeCity.ActiveReports.Samples.IListBinding.DataLayer
{
	/// <summary>
	/// Summary description for Product.
	/// </summary>
	public class Product
	{
		int _productID;
		string _productName;
		int _supplierID;
		int _categoryID;
		string _quantityPerUnit;
		decimal _unitPrice;
		int _unitsInStock;
		int _unitsOnOrder;
		int _reorderLevel;
		bool _discontinued;

		internal Product()
		{
		}

		//******
		//The properties listed below define a product in the datalayer. 
		//******

		public int ProductID
		{
			get
			{
				return _productID;
			}
			set
			{
				_productID = value;
			}
		}

		public string ProductName
		{
			get
			{
				return _productName;
			}
			set
			{
				_productName = value;
			}
		}

		public int SupplierID
		{
			get
			{
				return _supplierID;
			}
			set
			{
				_supplierID = value;
			}
		}

		public int CategoryID
		{
			get
			{
				return _categoryID;
			}
			set
			{
				_categoryID = value;
			}
		}


		public string QuantityPerUnit
		{
			get
			{
				return _quantityPerUnit;
			}
			set
			{
				_quantityPerUnit = value;
			}
		}


		public decimal UnitPrice
		{
			get
			{
				return _unitPrice;
			}
			set
			{
				_unitPrice = value;
			}
		}

		public int UnitsInStock
		{
			get
			{
				return _unitsInStock;
			}
			set
			{
				_unitsInStock = value;
			}
		}

		public int UnitsOnOrder
		{
			get
			{
				return _unitsOnOrder;
			}
			set
			{
				_unitsOnOrder = value;
			}
		}

		public int ReorderLevel
		{
			get
			{
				return _reorderLevel;
			}
			set
			{
				_reorderLevel = value;
			}
		}

		public bool Discontinued
		{
			get
			{
				return _discontinued;
			}
			set
			{
				_discontinued = value;
			}
		}
	}
}

File: DataBinding/Section/IListBinding/VB.NET/IListBinding.DataLayer/Product.vb
'<summary>
' Summary description for Product.
'</summary>
Public Class Product
	Private _productID As Integer
	Private _productName As String
	Private _supplierID As Integer
	Private _categoryID As Integer
	Private _quantityPerUnit As String
	Private _unitPrice As Decimal
	Private _unitsInStock As Integer
	Private _unitsOnOrder As Integer
	Private _reorderLevel As Integer
	Private _discontinued As Boolean

	Friend Sub New()
	End Sub 'New

	'******
	'* The properties listed below define a product in the datalayer. 
	'******

	Public Property ProductID() As Integer
		Get
			Return _productID
		End Get
		Set(ByVal Value As Integer)
			_productID = Value
		End Set
	End Property

	Public Property ProductName() As String
		Get
			Return _productName
		End Get
		Set(ByVal Value As String)
			_productName = Value
		End Set
	End Property

	Public Property SupplierID() As Integer
		Get
			Return _supplierID
		End Get
		Set(ByVal Value As Integer)
			_supplierID = Value
		End Set
	End Property

	Public Property CategoryID() As Integer
		Get
			Return _categoryID
		End Get
		Set(ByVal Value As Integer)
			_categoryID = Value
		End Set
	End Property

	Public Property QuantityPerUnit() As String
		Get
			Return _quantityPerUnit
		End Get
		Set(ByVal Value As String)
			_quantityPerUnit = Value
		End Set
	End Property

	Public Property UnitPrice() As Decimal
		Get
			Return _unitPrice
		End Get
		Set(ByVal Value As Decimal)
			_unitPrice = Value
		End Set
	End Property

	Public Property UnitsInStock() As Integer
		Get
			Return _unitsInStock
		End Get
		Set(ByVal Value As Integer)
			_unitsInStock = Value
		End Set
	End Property

	Public Property UnitsOnOrder() As Integer
		Get
			Return _unitsOnOrder
		End Get
		Set(ByVal Value As Integer)
			_unitsOnOrder = Value
		End Set
	End Property

	Public Property ReorderLevel() As Integer
		Get
			Return _reorderLevel
		End Get
		Set(ByVal Value As Integer)
			_reorderLevel = Value
		End Set
	End Property

	Public Property Discontinued() As Boolean
		Get
			Return _discontinued
		End Get
		Set(ByVal Value As Boolean)
			_discontinued = Value
		End Set
	End Property

End Class 'Product
